### ISSUE TITLE
Suggest using enum-map instead of HashMap with enum variant keys

### ISSUE THREAD

**Issue by leonardo-m** [rust-lang/rust-clippy#7329]
The naive_bytecount lint suggests to use a crate to perform an operation. So we could suggest to use a crate like:
https://crates.io/crates/enum-map

When code uses a HashMap with payload-free enum variants as keys, as here:
https://blog.frankel.ch/start-rust/2/

```rust
pub fn group_sidekicks_by_alignment<'a>(supers: &'a Vec<Super<'a>>) -> HashMap<Alignment, Vec<&'a Super<'a>>>  {
    let mut map = HashMap::new();                             
    map.insert(Good, Vec::new());                             
    map.insert(Evil, Vec::new());                             
    supers
        .iter()
        .filter(|&s| s.sidekick.is_some())                     
        .fold(map, |mut map, s| {                             
            let value = map.entry(s.alignment).or_default();  
            value.push(&s.sidekick.unwrap());                 
            map                                               
        })
}
```

(In Ada language this problems doesn't exist, because arrays are allowed to have enumeration keys).

**Comment by giraffate** [comment]
I don't know the details of adding `naive_bytecount`, but we're really careful not to suggest external crates.
